app/src/main/java/com/android/voyageur/ui/search/SearchUserProfile.kt

```diff
@@ -0,0 +1,101 @@
+package com.android.voyageur.ui.search
+
+import UserProfileContent
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import com.android.voyageur.model.user.User
+import com.android.voyageur.model.user.UserViewModel
+import com.android.voyageur.ui.navigation.NavigationActions
+import com.android.voyageur.ui.navigation.Route
+
+/**
+ * Composable function for displaying a detailed profile of a selected user. Displays a loading
+ * indicator if data is loading and navigates back to search if no user data is available.
+ *
+ * @param userViewModel The UserViewModel used for managing user data and interactions.
+ * @param navigationActions The NavigationActions used for handling navigation actions within the
+ *   app.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SearchUserProfileScreen(userViewModel: UserViewModel, navigationActions: NavigationActions) {
+  // Observing the selected user and loading state from the ViewModel
+  val user by userViewModel.selectedUser.collectAsState()
+  val isLoading by userViewModel.isLoading.collectAsState()
+
+  // Navigate back to search if no user data is available and loading is complete
+  if (user == null && !isLoading) {
+    LaunchedEffect(Unit) { navigationActions.navigateTo(Route.SEARCH) }
+    return
+  }
+
+  // Scaffold layout containing top bar for navigation and content for displaying user details
+  Scaffold(
+      modifier = Modifier.testTag("userProfileScreen"),
+      topBar = {
+        TopAppBar(
+            title = { Text("User Profile") },
```

Comment:
nitpick: from the screenshots I have seen this is shown above on the top of the screen but it is a general description, can we make it something like "Bogdan's profile". It is totally up to you though :)

Reply:
Great suggestion for enhancing the UX! I’ve implemented the change to display "Bogdan's profile" at the top of the screen. Thank you very much for the suggestion!